src/ClimaAtmos/Parameterizations/SurfaceFluxes/src/SurfaceFluxes.jl

```diff
@@ -0,0 +1,324 @@
+                                         γ_h,
+                                         β_m,
+                                         β_h,
+                                         Pr_0
+                                         )
+    logz = log(z_b/z_0)
+    zfactor = z_b/(z_b-z_0)*logz
+    s_b = Ri/Pr_0
+    if Ri > 0
+      temp = ((1-2*β_h*Ri)-sqrt(1+4*(β_h - β_m)*s_b))
+      ζ = zfactor/(2*β_h*(β_m*Ri - 1))*temp                    # Eq. 19 in Ref. Byun1990
+      L_mo = z_b/ζ                                             # LHS of Eq. 3 in Byun1990
+      ζ_0 = z_0/L_mo
+      ψ_m = ψ_m_stable(ζ, ζ_0, β_m)
+      ψ_h = ψ_h_stable(ζ, ζ_0, β_h)
+    else
+      Q_b = 1/9 * (1 /(γ_m^2) + 3 * γ_h/γ_m * s_b^2)           # Eq. 31 in Ref. Byun1990
+      P_b = 1/54 * (-2/(γ_m^3) + 9/γ_m * (-γ_h/γ_m + 3)*s_b^2) # Eq. 32 in Ref. Byun1990
+      crit = Q_b^3 - P_b^2
+      if crit < 0
+        T_b = cbrt(sqrt(-crit) + abs(P_b))                     # Eq. 34 in Ref. Byun1990
+        ζ = zfactor * (1/(3*γ_m)-(T_b + Q_b/T_b))              # Eq. 29 in Ref. Byun1990
+      else
+        θ_b = acos(P_b/sqrt(Q_b^3))                            # Eq. 33 in Ref. Byun1990
+        ζ = zfactor * (-2 * sqrt(Q_b) * cos(θ_b/3)+1/(3*γ_m))  # Eq. 28 in Ref. Byun1990
+      end
+      L_mo = z_b/ζ                                             # LHS of Eq. 3 in Byun1990
+      ζ_0 = z_0/L_mo
+      ψ_m = ψ_m_unstable(ζ, ζ_0, γ_m)
+      ψ_h = ψ_h_unstable(ζ, ζ_0, γ_h)
+    end
+    cu = k_Karman/(logz-ψ_m)                  # Eq. 10 in Ref. Byun1990, solved for u^*
+    cth = k_Karman/(logz-ψ_h)/Pr_0            # Eq. 11 in Ref. Byun1990, solved for h^*
+    C_D = cu^2                                                 # Eq. 36 in Byun1990
+    C_H = cu*cth                                               # Eq. 37 in Byun1990
+    return C_D, C_H, L_mo
+  end
+
+end # Byun1990 module
+
+module Nishizawa2018
+  using Utilities.RootSolvers
+  using Utilities.MoistThermodynamics
+  using PlanetParameters
+
+  """ Computes R_z0 expression, defined after Eq. 15 Ref. Nishizawa2018 """
+  compute_R_z0(z_0, Δz) = 1 - z_0/Δz
+
+  """ Computes f_m in Eq. A7 Ref. Nishizawa2018 """
+  compute_f_m(ζ) = (1-15*ζ)^(1/4)
+
+  """ Computes f_h in Eq. A8 Ref. Nishizawa2018 """
+  compute_f_h(ζ) = (1-9*ζ)^(1/2)
+
+  """ Computes ϕ_m in Eq. A1 Ref. Nishizawa2018 """
+  compute_ϕ_m(ζ, L, a) = L>=0 ? a*ζ+1 : (1-15*ζ)^(-1/4)
+
+  """ Computes ϕ_h in Eq. A2 Ref. Nishizawa2018 """
+  compute_ϕ_h(ζ, L, a, Pr) = L>=0 ? a*ζ/Pr+1 : (1-9*ζ)^(-1/2)
+
+  """ Computes ψ_m in Eq. A3 Ref. Nishizawa2018 """
+  function compute_ψ_m(ζ, L, a)
+    f_m = compute_f_m(ζ)
+    return L>=0 ? -a*ζ : log((1+f_m)^2*(1+f_m^2)/8) - 2*atan(f_m)+π/2
+  end
+
+  """ Computes ψ_h in Eq. A4 Ref. Nishizawa2018 """
+  compute_ψ_h(ζ, L, a, Pr) = L>=0 ? -a*ζ/Pr : 2*log((1+compute_f_h(ζ))/2)
+
+  """ Computes Ψ_m in Eq. A13 Ref. Nishizawa2018 """
+  compute_Ψ_m_low_ζ_lim(ζ, a) = ζ>=0 ? -a*ζ/2 : -15*ζ/8
```

Comment:
if this is not going to be called elsewhere, just roll it into the function (can add the eq number as a comment next to it)